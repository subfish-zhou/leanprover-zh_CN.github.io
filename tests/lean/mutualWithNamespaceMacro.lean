

mutual

inductive Foo.Lst (α : Type)
| nil   : Lst
| cons  : Tree α → Lst α → Lst α

inductive Boo.Tree (α : Type) -- conflicting namespace
| leaf : Tree α
| node : Lst α → Tree α

end

mutual

inductive Foo.Lst (α : Type)
| nil   : Lst α
| cons  : Tree α → Lst α → Lst α

inductive Foo.Tree (α : Type)
| leaf : Tree α
| node : Lst α → Tree α

end
