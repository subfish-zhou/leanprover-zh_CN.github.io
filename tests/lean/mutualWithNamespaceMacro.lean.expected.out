mutualWithNamespaceMacro.lean:9:0-11:23: error: conflicting namespaces in mutual declaration, using namespace 'Boo', but used 'Foo' in previous declaration
